File: frontend/src/microservices/train/objects/TrainAdminSessionPlannerResponse.ts
// 本文件定义 train 模块的 `TrainAdminSessionPlannerResponse`，用于表示铁路管理员会话并提供 JSON 编解码。

import type { TrainPlannerResponse } from './TrainPlannerResponse'

export type TrainAdminSessionPlannerResponse = {
  managerId: string
  operatorCode: string
  email: string
  displayName: string
  status: string
  managedTrains: TrainPlannerResponse[]
}
export const trainAdminSessionPlannerResponseFromJson = (json: string): TrainAdminSessionPlannerResponse =>
  JSON.parse(json) as TrainAdminSessionPlannerResponse

export const trainAdminSessionPlannerResponseToJson = (value: TrainAdminSessionPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/operations/train/objects/TrainAdminSessionPlannerResponse.scala
package com.typesafe.travel.operations.domain

import com.typesafe.travel.train.domain.TrainPlannerResponse
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TrainAdminSessionPlannerResponse(
    managerId: String,
    operatorCode: String,
    email: String,
    displayName: String,
    status: String,
    managedTrains: List[TrainPlannerResponse]
)
object TrainAdminSessionPlannerResponse:
  given sourceEncoder: Encoder[TrainAdminSessionPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[TrainAdminSessionPlannerResponse] = deriveDecoder
